jwala-services/src/main/java/com/cerner/jwala/service/resource/impl/handler/GroupLevelAppResourceHandler.java

```diff
@@ -63,51 +72,74 @@ public ConfigTemplate fetchResource(final ResourceIdentifier resourceIdentifier)
-                    metaData.getDeployFileName(), metaData.getJsonData(), templateContent);
-
-            // Can't we just get the application using the group name and target app name instead of getting all the applications
-            // then iterating it to compare with the target app name ???
-            // If we can do that then TODO: Refactor this to return only one application and remove the iteration!
-            final List<Application> applications = applicationPersistenceService.findApplicationsBelongingTo(groupName);
-
-            for (final Application application : applications) {
-                if (metaData.getEntity().getDeployToJvms() && application.getName().equals(resourceIdentifier.webAppName)) {
-                    for (final Jvm jvm : group.getJvms()) {
-                        UploadAppTemplateRequest uploadAppTemplateRequest = new UploadAppTemplateRequest(application, metaData.getTemplateName(),
-                                metaData.getDeployFileName(), jvm.getJvmName(), metaData.getJsonData(), templateContent
-                        );
-                        JpaJvm jpaJvm = jvmPersistenceService.getJpaJvm(jvm.getId(), false);
-                        applicationPersistenceService.uploadAppTemplate(uploadAppTemplateRequest, jpaJvm);
-                    }
-                }
-            }
+                    metaDataCopy.getDeployFileName(), metaDataCopy.getJsonData(), templateContent);
+
+            createJvmTemplateFromAppResource(resourceIdentifier, templateContent, metaDataCopy, groupName, group);
 
             createResourceResponseWrapper = new CreateResourceResponseWrapper(createdConfigTemplate);
         } else if (successor != null) {
-            createResourceResponseWrapper = successor.createResource(resourceIdentifier, metaData, templateContent);
+            createResourceResponseWrapper = successor.createResource(resourceIdentifier, metaDataCopy, templateContent);
         }
         return createResourceResponseWrapper;
     }
 
+    private ResourceTemplateMetaData updateApplicationWarMetaData(ResourceIdentifier resourceIdentifier, ResourceTemplateMetaData metaDataCopy, Application app) {
+        boolean isUnpackWar = app.isUnpackWar();
+        metaDataCopy = new ResourceTemplateMetaData(metaDataCopy.getTemplateName(), metaDataCopy.getContentType(), metaDataCopy.getDeployFileName(), metaDataCopy.getDeployPath(), metaDataCopy.getEntity(), isUnpackWar, metaDataCopy.isOverwrite());
+
+        try {
+            final ObjectMapper objectMapper = new ObjectMapper();
+            String jsonData = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(metaDataCopy);
+            metaDataCopy.setJsonData(jsonData);
+        } catch (IOException e) {
+            final String errMsg = MessageFormat.format("Unable to convert meta data to a string for app {0} creating resource {1}", resourceIdentifier.webAppName, metaDataCopy.getDeployFileName(), e);
```

Comment:
For me the message that would be easier to understand in a troubleshooting scenario is the following

    Failed to update the war meta data for app {0} on {1} resource creation

The reason is because at first glance it would call my attention immediately on "update war meta data" and then for the details, I'd just read the stack trace.

Anyways this is a minor thing and might be a personal preference. The team might have another opinion.